src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -4,35 +4,108 @@
+        commonSecretLabels.put(Utils.MANAGED_BY_LABEL_KEY, Utils.KAFKA_ACCESS_LABEL_VALUE);
     }
 
     @Override
-    public UpdateControl<KafkaAccess> reconcile(KafkaAccess kafkaAccess, Context context) {
-        return UpdateControl.noUpdate();
+    public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final Context context) {
+        final String kafkaAccessName = kafkaAccess.getMetadata().getName();
+        LOGGER.info("Reconciling KafkaAccess " + kafkaAccessName);
+        final Map<String, String> data  = new HashMap<>(commonSecretData);
+        kubernetesClient
+            .secrets()
+            .inNamespace(kafkaAccess.getMetadata().getNamespace())
+            .createOrReplace(
+                    new SecretBuilder()
+                        .withType(SECRET_TYPE)
+                        .withNewMetadata()
+                        .withName(kafkaAccessName)
+                        .withLabels(commonSecretLabels)
+                        .withOwnerReferences(
+                                new OwnerReferenceBuilder()
+                                    .withApiVersion(kafkaAccess.getApiVersion())
+                                    .withKind(kafkaAccess.getKind())
+                                    .withName(kafkaAccessName)
+                                    .withUid(kafkaAccess.getMetadata().getUid())
+                                    .withBlockOwnerDeletion(false)
+                                    .withController(false)
+                                    .build()
+                        )
+                        .endMetadata()
+                        .withData(data)
+                        .build()
+            );
+        final KafkaAccessStatus status = Optional.ofNullable(kafkaAccess.getStatus())
+                .orElse(new KafkaAccessStatus());
+        status.setBinding(new BindingStatus(kafkaAccessName));
+        kafkaAccess.setStatus(status);
+        return UpdateControl.updateStatus(kafkaAccess);
     }
 
     @Override
-    public List<EventSource> prepareEventSources(EventSourceContext<KafkaAccess> context) {
-        return Collections.emptyList();
+    public List<EventSource> prepareEventSources(final EventSourceContext<KafkaAccess> context) {
+        LOGGER.info("Preparing event sources");
+        final List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(new InformerEventSource<>(kubernetesClient,
+                Kafka.class,
+                kafka -> Utils.getKafkaAccessResourceIDsForKafkaInstance(context.getPrimaryCache().list(), kafka)
+        ));
+        final SharedIndexInformer<Secret> secretInformer =
+                kubernetesClient.secrets().inAnyNamespace()
+                        .withLabelIn(Utils.MANAGED_BY_LABEL_KEY, Utils.STRIMZI_CLUSTER_LABEL_VALUE, Utils.KAFKA_ACCESS_LABEL_VALUE)
+                        .runnableInformer(0);
+        eventSources.add(new InformerEventSource<>(secretInformer,
```

Comment:
Not sure if I understand this filtering part but, if you just expect to have one secret there (not more), then usually it enough to do as it is here:
https://github.com/java-operator-sdk/java-operator-sdk/pull/841/files
(added yesterday, you have to use 2.0.1) Please let me know if this is not enough in you case.

Reply:
Ok see that there are multiple cases, so the trivial mapper is probably not enough.

Reply:
I wonder if I could use that mapper to at least get the secret that KafkaAccess created? But yeah I also need to check some other resources that the operator didn't create.

Reply:
Hmm, will take a look, little bit modify how it is there, so you are able to use that mapper for different situations.

Reply:
https://github.com/java-operator-sdk/java-operator-sdk/pull/846/files